### PR TITLE
Adds test for buildAnswer() with JSON.

### PR DESCRIPTION
**Pull request by abundis28**
# What does this PR do?
- Adds test for buildQuestion().

**Related Issues**
- 

---
**Merge Checklist**
- [ ] Branch is up to date
- [ ] No code style issues

### CODE REVIEW

File: src/test/java/com/google/sps/UtilityTest.java [hunk 1]
```diff
@@ -623,4 +623,52 @@ public void executeQuery_incorrectQuery_doesNothing() {
...
+    ResultSet resultSetMock = mock(ResultSet.class);
+    try {
+      // Mock behaviors for answer creation.
+      when(resultSetMock.getInt(SqlConstants.ANSWER_FETCH_ID)).thenReturn(1);
+      when(resultSetMock.getString(SqlConstants.ANSWER_FETCH_BODY)).thenReturn(
+          "You may get it at your college!");
+      when(resultSetMock.getString(SqlConstants.ANSWER_FETCH_AUTHORNAME)).thenReturn(
+          "Andres Abundis");
+      when(resultSetMock.getTimestamp(SqlConstants.ANSWER_FETCH_DATETIME)).thenReturn(
+          new Timestamp(1598899890));
```
Comment (timothy-parsons) [inline src/test/java/com/google/sps/UtilityTest.java:639]: When you reference values multiple times in your test, it's probably best to make them into variables at the top of your test and then reference those variables throughout the test instead for the sake of readability!

File: src/test/java/com/google/sps/UtilityTest.java [hunk 2]
```diff
@@ -623,4 +623,52 @@ public void executeQuery_incorrectQuery_doesNothing() {
...
+    expectedAnswer.setAuthorName("Andres Abundis");
+    expectedAnswer.setDateTime(new Timestamp(1598899890));
+    // Insert previously created expectedComment to expectedAnswer.
+    expectedAnswer.addComment(expectedComment);
+
+    Assert.assertEquals(Utility.convertToJsonUsingGson(expectedAnswer), 
+                        Utility.convertToJsonUsingGson(actualAnswer));
+                        
+    // Asserting the equality of JSONs because of trouble with the EqualsBuilder.
+    // Assert.assertTrue(EqualsBuilder.reflectionEquals(expectedAnswer,actualAnswer));
```
Comment (timothy-parsons) [inline src/test/java/com/google/sps/UtilityTest.java:672]: Bummer - you can remove this commented out code and instead add a TODO to update the test to utilize the EqualsBuilder!

File: src/test/java/com/google/sps/UtilityTest.java [hunk 3]
```diff
@@ -623,4 +623,52 @@ public void executeQuery_incorrectQuery_doesNothing() {
 
     Assert.assertTrue(notificationId == 0);
   }
+
+  /** Tests function buildAnswerFull(). */
+  @Test
+  public void buildAnswer_validResultSet_successfulBuild() {
```
Comment (timothy-parsons) [inline src/test/java/com/google/sps/UtilityTest.java:629]: This test looks good - think through other test cases you might support. For instance, when SQL columns are missing, or the comments are empty, or the answer is empty, etc. You'll want to be able to look at your test names and have confidence that all the corner cases of the functionality is accounted for.

**Review by timothy-parsons — APPROVED** [review]
It's late on your last day so I'm not going to push for the changes I suggested, instead think through them for a few minutes!

**Review by shaargtz — APPROVED** [review]

File: src/test/java/com/google/sps/UtilityTest.java (hunk 2 above)
Comment (abundis28) [reply src/test/java/com/google/sps/UtilityTest.java:672]: Ack!

File: src/test/java/com/google/sps/UtilityTest.java (hunk 3 above)
Comment (abundis28) [reply src/test/java/com/google/sps/UtilityTest.java:629]: Will think through them!
